crates/electrum/src/electrum_ext.rs

```diff
@@ -118,7 +108,7 @@ impl<K> ElectrumUpdate<K, ConfirmationHeightAnchor> {
             .collect::<HashMap<u32, u64>>();
 
         let graph_changeset = {
-            let old_changeset = TxGraph::default().apply_update(update.graph.clone());
+            let old_changeset = TxGraph::default().apply_update(self.graph_update);
```

Comment:
Not part of this PR necessarily but the code below seems super complicated and uses changesets for no reason? How about we just have a function on `tx_graph.map_anchor(|anchor| get_new_anchor(anchor))` which consumes self and returns a new `TxGraph` with the new anchor.

Reply:
Created issue: #1096